src/classrooms/classrooms.service.ts

```diff
@@ -56,4 +56,36 @@ export class ClassroomsService {
       userId: user.id,
     });
   }
+
+  async getClassroomsForUser(userId: number): Promise<ClassroomResponseDto[]> {
+    const user = await this.em.findOne(User, { id: userId }, { populate: ["teacher"] });
+    if (!user) {
+      throw new NotFoundException("User not found");
+    }
+
+    if (!user.teacher) {
+      throw new ForbiddenException("Only teachers can view classrooms");
+    }
+
+    const classrooms = await this.em.find(
+      Classroom,
+      { teacher: user.teacher },
+      {
+        orderBy: { classTime: "DESC" },
+        populate: ["teacher"],
+      },
+    );
+
+    return classrooms.map(
+      (classroom) =>
+        new ClassroomResponseDto({
+          id: classroom.id,
+          title: classroom.title,
+          subject: classroom.subject,
+          classTime: classroom.classTime,
+          days: classroom.days,
+          userId: user.id,
+        }),
+    );
```

Comment:
Is this doing what a serializer should do?

Reply:
Yes bhaiya, it's taking the raw classroom entity and transforming it into a response format suitable for frontend/client.

Reply:
Mikro ORM provides a serialize method to do this. You can use that.

And serializer should be called at the controller level imported from classrooms.serializer.ts.